great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -1360,20 +1362,27 @@ def get_batch_data_and_markers(
         create_temp_table: bool = batch_spec.get(
             "create_temp_table", self._create_temp_table
         )
+        # this is where splitter components are added to the selectable
```

Comment:
I think the return type of this function can be updated to: `Tuple[SqlAlchemyBatchData | None, BatchMarkers]`